k8s-openapi-codegen-common/src/templates/struct_deep_merge.rs

```diff
@@ -13,11 +15,43 @@ pub(crate) fn generate(
 	let type_generics_where = generics.where_part.map(|part| format!(" where {part}")).unwrap_or_default();
 
 	let mut merge_body = String::new();
-	for super::Property { field_name, .. } in fields {
-		writeln!(
-			&mut merge_body,
-			"        {local}DeepMerge::merge_from(&mut self.{field_name}, other.{field_name});",
-		)?;
+	for super::Property { field_name, merge_type, .. } in fields {
+		match merge_type {
```

Comment:
Sadly this breaks the expectation that `foo.bar.merge_from(baz.bar)` will work as a subset operation of `foo.merge_from(baz)`. An alternative (and probably one I'd prefer if this was greenfield) would be to provide newtypes for the different `Vec`/`BTreeMap` variants, but that means a pretty large amount of breakage even for users who don't care about `DeepMerge` at all (the vast majority, I suspect).

Reply:
I was thinking that we would define newtype per strategy (`VecSet`, `VecMap`, etc).

Reply:
Yes, that would work too. But also I think it would be easier to just remove the `DeepMerge` impls for `Vec` and `BTreeMap`, so that the user is forced to choose one of the merge strategies.

Reply:
It looks like we have a few cases of `BTreeMap<_, Vec<_>>` which would be broken by this, such as [`SubjectAccessReviewSpec.extra`](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.23/#subjectaccessreviewspec-v1-authorization-k8s-io).

Reply:
Done in f876657c5cd9a6773c94345ec9c9172b52cd430b, bb57d5b4486ec9104b94143b2b8fc9b86627bc36.